src/lib/configs/project/spell.ts

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { access } from "fs/promises";
+import { dirname, resolve } from "path";
+
+import type { JSONSchemaType } from "ajv";
+
+import {
+  FLUENCE_CONFIG_FILE_NAME,
+  SPELL_CONFIG_FILE_NAME,
+  TOP_LEVEL_SCHEMA_ID,
+  U32_MAX,
+} from "../../const.js";
+import { ensureSpellAbsolutePath } from "../../helpers/downloadFile.js";
+import { validateBatch } from "../../helpers/validations.js";
+import { ensureFluenceDir } from "../../paths.js";
+import {
+  getConfigInitFunction,
+  InitConfigOptions,
+  InitializedConfig,
+  InitializedReadonlyConfig,
+  getReadonlyConfigInitFunction,
+  Migrations,
+  ConfigValidateFunction,
+} from "../initConfig.js";
+
+const MAX_PERIOD_YEAR = 100;
+
+/** Max period is 100 years in secs: 60 sec * 60 min * 24 hours * 365 days * 100 years */
+const MAX_PERIOD_SEC = 60 * 60 * 24 * 365 * MAX_PERIOD_YEAR;
+
+type SpellProperties = {
+  aquaFilePath: string;
+  function: string;
+  initArgs?: Record<string, unknown>;
+  clock?: {
+    periodSec?: number;
+    startTimestamp?: string;
+    endTimestamp?: string;
+    startDelaySec?: number;
+    endDelaySec?: number;
+  };
+};
+
+export type OverridableSpellProperties = Partial<SpellProperties>;
+
+export type ConfigV0 = {
+  version: 0;
+} & SpellProperties;
+
+const spellProperties = {
+  version: { type: "number", const: 0 },
+  aquaFilePath: {
+    type: "string",
+    description:
+      "Path to Aqua file which contains an Aqua function that you want to use as a spell",
+  },
+  function: {
+    type: "string",
+    description: "Name of the Aqua function that you want to use as a spell",
+  },
+  initArgs: {
+    type: "object",
+    description:
+      "A map of Aqua function arguments names as keys and arguments values as values. They will be passed to the spell function. Also they will be stored in the key-value storage for this particular spell. It's possible to change argument values inside the spell itself if you return a new map of new arguments from the spell function. Also it's possible to update these values from another spell using Spell.set_string method",
```

Comment:
```suggestion
      "A map of Aqua function arguments names as keys and arguments values as values. They will be passed to the spell function and will be stored in the key-value storage for this particular spell. It's possible to change argument values inside the spell itself if you return a new map of new arguments from the spell function or using Spell.set_string method",
```

Reply:
But maybe explaining how to modify these arguments isn't a good idea here. I think this explanation might be confusing without examples. 

```suggestion
      "A map of Aqua function arguments names as keys and arguments values as values. They will be passed to the spell function and will be stored in the key-value storage for this particular spell.",
```